pubnub-core/src/subscription/subscribe_loop.rs

```diff
@@ -76,32 +72,20 @@ where
 
         transport,
 
-        origin,
-        agent: _, // TODO: use it.
-        subscribe_key,
-
         mut channels,
-        mut groups,
+        mut channel_groups,
     } = params;
 
-    let mut encoded_channels = EncodedChannelsList::from(&channels);
-    let mut encoded_groups = EncodedChannelsList::from(&groups);
-
     let mut timetoken = Timetoken::default();
 
     loop {
-        // TODO: move the cache into the transport once we move from passing the
-        // URL to a structured transport invocation.
-        let request_encoded_channels = encoded_channels.clone();
-        let response = subscribe_request::subscribe_request(
-            &transport,
-            subscribe_request::SubscribeRequestParams {
-                origin: &origin,
-                subscribe_key: &subscribe_key,
-                encoded_channels: &request_encoded_channels,
-            },
-            timetoken.clone(),
-        );
+        // TODO: re-add cache.
```

Comment:
👍 This is also where we want to pass the list of groups.

Reply:
Yes, I've already added proper groups support in my next iteration.